src/pages/JoinStep2/JoinStep2.js

```diff
@@ -0,0 +1,365 @@
+import React, { Component } from "react";
+import { Link } from "react-router-dom";
+import "./JoinStep2.scss";
+
+class JoinStep2 extends Component {
+  constructor() {
+    super();
+    this.state = {
+      name: "",
+      account: "",
+      password: "",
+      email: "",
+      phoneNumber: "",
+      dateOfBirth: 19951116,
+      emailAgreement: false,
+      smsAgreement: false,
+      //여기서부터 백에 전달 안함
+      confirmPw: "",
+      isNameNull: true,
+      isAccountNull: true,
+      isPasswordNull: true,
+      isEmailNull: true,
+      isPhoneNumberNull: true
+    };
+  }
+
+  handleInput = e => {
+    e.preventDefault();
+    const { value, name } = e.target;
+    this.setState({
+      [name]: String(value)
+    });
+    console.log(this.state);
+  };
+
+  handleSMSCheckbox = e => {
+    const { name } = e.target;
+    this.setState({
+      [name]: !this.state.smsAgreement
+    });
+    console.log(this.state);
+  };
+
+  handleEmailCheckbox = e => {
+    const { name } = e.target;
+    this.setState({
+      [name]: !this.state.emailAgreement
+    });
+    console.log(this.state);
+  };
+
+  checkId = () => {
+    const APIforIdCheck = "http://localhost:3000/data/account.json";
+    const { account } = this.state;
+    fetch(APIforIdCheck, {
+      method: "POST",
+      body: JSON.stringify({
+        account: account
+      })
+    })
+      .then(response => response.json())
+      .then(result => {
+        console.log("백엔드에서 오는 응답 메세지:", result);
+      });
+  };
+
+  checkEmail = () => {
+    const APIforEmailCheck = "http://localhost:3000/data/email.json";
+    const { email } = this.state;
+    fetch(APIforEmailCheck, {
+      method: "POST",
+      body: JSON.stringify({
+        email: email
+      })
+    })
+      .then(response => response.json())
+      .then(result => {
+        console.log("백엔드에서 오는 응답 메세지:", result);
+      });
+  };
+
+  isPwInfoValid = password => {
+    let number = /[0-9]/g;
+    let character = /[a-zA-z]/g;
+    let specialCharacter = /\W|_/g;
+    const isNumIncluded = password.match(number) ? true : false;
```

Comment:
```suggestion
    const isNumIncluded = !!password.match(number);
```
- true일 때 true를, false일 때 false를 리턴하는 경우 굳이 삼항연산자를 써주지 않아도 될 것 같네요
- boolean 값임을 표현하는 방법은 사용해주신 것처럼 변수명을 `is-`로 시작하시거나 값에 부정연산자를 두번 사용하는 경우 등이 있겠네요